authentication/views.py

```diff
@@ -788,3 +791,65 @@ def post(self, request, **args):
             },
             status=status.HTTP_200_OK
         )
+
+
+class SavedCardsListView(generics.GenericAPIView):
+    """get all a  cards for authenticated user"""
+
+    permission_classes = (IsAuthenticated,)
+    serializer_class = CardInfoSerializer
+    def get_queryset(self):
+
+        user = self.request.user
+        return CardInfo.objects.filter(user_id=user.id)
+
+    def get(self, request):
+        serializer = self.serializer_class(self.get_queryset(), many=True)
+        if not serializer.data:
+            response = {
+                "data": {
+                    "message": "You don't have any card saved"
+                }
+            }
+            status_code = status.HTTP_404_NOT_FOUND
+        else:
+            response = {
+                "data": {
+                    "saved_card(s)": serializer.data,
+                    "message": "Cards retrieved",
+                }
+            }
+            status_code = status.HTTP_200_OK
+        return Response(response, status=status_code)
+
+
+class DeleteSavedCardView(generics.GenericAPIView):
+    """ Remove existing card """
+
+    permission_classes = (IsAuthenticated, IsCardOwner)
+
+    def get_object(self, request, card_info_id):
+        """
+        Handles getting a specific card
+
+        :param request:
+        :param user_id:
+        :return: Object
+        """
+        try:
+            card = CardInfo.objects.get(id=card_info_id)
+            self.check_object_permissions(request, card)
```

Comment:
Hey @Niyitangasam , kindly clarify what this line does

Reply:
Hey @Tevinthuku, thanks for reminding me, I just removed it because I already checked the permission and there is no need to enforce it.